src/data/real/nnreal.lean

```diff
@@ -348,13 +348,16 @@ begin
   { simp [sup_eq_max, max_eq_left h, max_eq_left (mul_le_mul_of_nonneg_left h (zero_le a))] },
 end
 
-lemma mul_finset_sup {α} {f : α → ℝ≥0} {s : finset α} (r : ℝ≥0) :
-  r * s.sup f = s.sup (λa, r * f a) :=
-begin
-  refine s.induction_on _ _,
-  { simp [bot_eq_zero] },
-  { assume a s has ih, simp [has, ih, mul_sup], }
-end
+lemma sup_mul (a b c : ℝ≥0) : (a ⊔ b) * c = (a * c) ⊔ (b * c) :=
```

Comment:
Can you add that to `algebra.order.lattice_group` next to `mul_sup`? and delete the lemma above as well.

Reply:
The lemma in `lattice_group` doesn't apply here, because `ℝ≥0` is not a `comm_group`.

Reply:
Can you generalize `mul_sup` to linear ordered monoids then?

Reply:
I don't want to go down yet another rabbit hole, and would prefer to declare that out of scope for this PR.

Reply:
Use `mul_max_of_nonneg` at least then.

Reply:
Good idea, done.